### PR TITLE
[CI] Make the linux-modules-extras conditional

### PR DESCRIPTION
**Pull request by zdohnal**
Several runs fail on sound dummy part, check if this works...

### CODE REVIEW

**Comment by chrisbra** [comment]
Hm, why did this not run any github CI Actions?

**Comment by zdohnal** [comment]
@chrisbra I'm confused myself :D

**Comment by zdohnal** [comment]
@chrisbra ok, my bad https://github.com/vim/vim/actions/runs/5864529562 - looks like {{ }} is only for yaml substitutions.

**Comment by chrisbra** [comment]
relevant github issue: https://github.com/actions/runner-images/issues/8080

**Comment by zdohnal** [comment]
But still the CI should go the report it is missing and skip... unless apt returns 0 if it doesn't find package :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -180,24 +180,26 @@ jobs:
           sudo usermod -a -G audio "${USER}"
           sudo bash ci/setup-xvfb.sh
 
-      - name: Set up snd-dummy
+      - name: Check if linux-modules-extra is available
+        id: check-snd-dummy
         if: (!(contains(matrix.extra, 'unittests') || contains(matrix.extra, 'vimtags')))
```
Comment (jamacku) [inline .github/workflows/ci.yml:185]: You can try this condition:

```suggestion
        if: (!(contains(matrix.extra, 'unittests') || contains(matrix.extra, 'vimtags'))) && runner.name != 'ubuntu-22.04'
```

[runner context](https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context)

**Comment by pheiduck** [comment]
Nice, but I would prefer caching of the package like I try to get ready here:
https://github.com/vim/vim/pull/12779

**Comment by zdohnal** [comment]
> Nice, but I would prefer caching of the package like I try to get ready here: #12779

@pheiduck  That's a good idea - however the current problem is that package does not install at all in the container,  which makes CI fail (and people have to check the errors...), so unless you want to store the package on your own mirror and maintain it, caching does nothing at the moment for CI errors.

Caching of the downloaded packages can be applied on this PR once this PR is merged. (basically you merge the steps again and use only continue-on-error - maybe some additional conds in next steps will be needed too).

**Comment by pheiduck** [comment]
@zdohnal yep, I know, I will continue working on this when the Package is available again.
Rebased to this PR when merged.

**Comment by zdohnal** [comment]
@chrisbra macos test passed before, I don't see a reaso why it should fail now https://github.com/vim/vim/actions/runs/5866102553/job/15904274375?pr=12801 .

**Pull request closed without merging**